### PR TITLE
improvement: thresholds for folding regions

### PR DESCRIPTION
**Pull request by kasiaMarek**
- don't merge region with parent when if parent too small without it
- allow to configure folding region threshold using server properties
- make threshold work consistently

resolves: https://github.com/scalameta/metals/issues/6938

### CODE REVIEW

File: tests/unit/src/test/resources/foldingRange-scala3-foldLineOnly/expect/Main.java [hunk 1]
```diff
@@ -41,13 +42,13 @@ String hello(String str) >>region>>{
             return "asssd";
         }<<region<<
 
-        void openingCommentInStr() >>region>>{
```
Comment (tgodzik) [inline tests/unit/src/test/resources/foldingRange-scala3-foldLineOnly/expect/Main.java:44]: Why did we lose a region here? That should be 3 lines with all the {}

Same below

File: tests/unit/src/test/resources/foldingRange-scala3-foldLineOnly/expect/i6938.scala [hunk 2]
```diff
@@ -0,0 +1,28 @@
+def xd =>>region>>
+  val t =
+    1 + 2
+
+  t match
```
Comment (tgodzik) [inline tests/unit/src/test/resources/foldingRange-scala3-foldLineOnly/expect/i6938.scala:5]: We should have a region for match as weel

File: tests/unit/src/test/resources/foldingRange/expect/ForLoop.scala [hunk 3]
```diff
@@ -7,14 +7,14 @@ class A >>region>>{
     }<<region<<
 
   def noSpacing =>>region>>
-    for>>region>>{
```
Comment (tgodzik) [inline tests/unit/src/test/resources/foldingRange/expect/ForLoop.scala:10]: We should still have this region as well, no?

File: tests/unit/src/test/resources/foldingRange-scala3-foldLineOnly/expect/Main.java (hunk 1 above)
Comment (kasiaMarek) [reply tests/unit/src/test/resources/foldingRange-scala3-foldLineOnly/expect/Main.java:44]: I counted only the lines that actually get folded, so:
```scala
def a = {
  1
}
```
is one
```scala
def a =
  1
```
is also one. Think it's a bit more intuitive than if these had two and one respectively. I can offset by one, so both count as 2, would that be better?

Comment (tgodzik) [reply tests/unit/src/test/resources/foldingRange-scala3-foldLineOnly/expect/Main.java:44]: ok, let's try this solution. I don't really have preference and it looks sensible. Thanks for explaining!

**Review by tgodzik — APPROVED** [review]
LGTM!

**Comment by KacperFKorban** [comment]
Thanks! This seems great :heart:

**Pull request merged**